dts/binding-template.yaml

```diff
@@ -154,6 +166,31 @@ properties:
         required: true
         const: 1
 
+    int-with-default:
+        type: int
+        required: false
+        default: 123
+
+    array-with-default:
+        type: array
+        required: false
+        default: [1, 2, 3] # Same as 'array-with-default = <1 2 3>'
+
+    string-with-default:
+        type: string
+        required: false
+        default: "foo"
+
+    string-array-with-default:
+        type: string-array
+        required: false
+        default: ["foo", "bar"] # Same as 'string-array-with-default = "foo", "bar"'
+
+    uint8-array-with-default:
+        type: uint8-array
+        required: false
+        default: [0x12, 0x34] # Same as 'uint8-array-with-default = [12 34]'
+
```

Comment:
no `boolean-with-default` ?
I see you took care of it in the lib, so this comment would just be about documenting it.

Reply:
If it's handled, it's unintentional. ;)

Putting `default:` on a `boolean` gives an error:

```python
if prop_type in {"boolean", "compound", "phandle", "phandles",
                 "phandle-array"}:
    _err("'default:' can't be combined with 'type: {}' for '{}' in "
         "'properties:' in {}".format(prop_type, prop_name, binding_path))
```

Not sure what it would mean to put a default on a boolean. `default: true` seems weird, because then there'd be no way to check if the property is there (would get `true` both if it's there and if it's missing), and `default: false` would be a no-op, since that's already the value if  the property is missing.

Note that the `default:` value has to match the type.

Reply:
Yeah, this fits in with my comment above.  `boolean` by its very model is always provided, as either 1 or 0.  Functionally it's as if `boolean` had an implicit `required` with implicit default `false`.  Only when a property is `optional` do you fail to get a generated symbol for it.

Reply:
@erwango 
Ping

Anything about the current behavior you're unhappy with?

Reply:
@ulfalizer, my point was to care about `default: false`, but indeed that's already taken care of, so fine with me! Txs